plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/LibvirtDomainXMLParser.java

```diff
@@ -171,6 +174,25 @@ public boolean parseDomainXML(String domXML) {
                 interfaces.add(def);
             }
 
+            NodeList ports = devices.getElementsByTagName("channel");
+            for (int i = 0; i < ports.getLength(); i++) {
+                Element channel = (Element)ports.item(i);
+
+                String type = channel.getAttribute("type");
+                String path = getAttrValue("source", "path", channel);
+                String name = getAttrValue("target", "name", channel);
+                String state = getAttrValue("target", "state", channel);
+
+                ChannelDef def = null;
+                if (Strings.isNullOrEmpty(state)) {
+                    def = new ChannelDef(name, ChannelDef.ChannelType.valueOf(type.toUpperCase()), path);
+                } else {
+                    def = new ChannelDef(name, ChannelDef.ChannelType.valueOf(type.toUpperCase()), ChannelDef.ChannelState.valueOf(state.toUpperCase()), path);
```

Comment:
What if ``state`` is ``null``?  While it likely shouldn't happen, malformed XML could cause an NPE which will not be helpful to operators or developers trying to debug the issue.  Consider adding a ``checkState(StringUtils.isNoneBlank(state), "<explanation of unexpected condition>")`` on line 185.